### PR TITLE
feat: adds shopify-cart to gatsby reference store

### PR DESCRIPTION
**Pull request by mdarrik**
## WHY are these changes introduced?

Replaces Shopify Checkout with Shopify Cart in the Gatsby reference store [ENG-7065](https://nacelle.atlassian.net/browse/ENG-7065) 

## WHAT is this pull request doing?

- Replaces Shopify Checkout with Shopify Cart
- adds an error modal using the native dialog element 

## How to Test

1. Checkout this branch `mdarrik/ENG-7065-shopify-cart-gatsby-reference-store`
2. Run `npm run bootstrap` from the root of the repo && `cd reference-stores/gatsby`
3. add the new environment variables to your .env file as specified in .env.example
4. test local dev with `npm run develop`
5. Test a full build with `npm run build && npx serve public/`

### CODE REVIEW

File: reference-stores/gatsby/src/components/Error/ErrorModal.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                    focus:ring-offset-2
+                    sm:ml-3 sm:w-auto sm:text-sm
+                  "
+          onClick={closeErrorModal}
+        >
+          Close
+        </button>
+      </div>
+      {/* </div>
+      </div> */}
```
Comment (NWRichmond) [inline reference-stores/gatsby/src/components/Error/ErrorModal.js:135]:
```suggestion
```

File: reference-stores/gatsby/src/components/Error/ErrorModal.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            transform
+            overflow-hidden
+            rounded-lg
+            bg-white
+            text-left
+            shadow-xl
+            transition-all
+            sm:my-8 sm:w-full sm:max-w-lg
+          "
+        > */}
```
Comment (NWRichmond) [inline reference-stores/gatsby/src/components/Error/ErrorModal.js:56]:
```suggestion
```

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Removing items from the cart & checkout is working as expected ✅

There's an issue with add-to-cart on https://nacelle-gatsby-reference-store-git-mdarrik-eng-7-b4c0ca-nacelle.vercel.app that isn't present in [the Next.js Reference Store with Shopify Cart](https://nacelle-next-reference-store-git-eng-6843-shopif-ccc911-nacelle.vercel.app) - when adding an item to the cart:
- The item doesn't appear in the cart optimistically
- The first item in the cart is incremented

Both issues are resolved once the cart is updated with the Shopify Cart response.

https://user-images.githubusercontent.com/5732000/196734648-12343f8b-79a7-4870-8c59-0d2aa463930f.mov

Once this issue is resolved, I think this PR will be good to go!

**Review by NWRichmond — APPROVED** [review]
Thanks for those updates, @mdarrik! Everything is working as expected on https://nacelle-gatsby-reference-store-git-mdarrik-eng-7-b4c0ca-nacelle.vercel.app :)

https://user-images.githubusercontent.com/5732000/196788957-554d5306-4f34-4fa8-970a-13674bdf082d.mov

**Review by KrisQ — APPROVED** [review]
Looks good!

**Pull request merged**